### PR TITLE
Fix dependency to use SHA's of local files, not cached shas 

### PR DESCRIPTION
**Pull request by rcuhljr**
Also some rubocop fixes.

### CODE REVIEW

File: dependency.lic [hunk 1]
```diff
@@ -305,12 +306,6 @@ class ScriptManager
     end
   end
 
-  def load_script(filename, force = false)
```
Comment (rcuhljr) [inline dependency.lic]: Unused.

File: dependency.lic [hunk 2]
```diff
@@ -875,10 +861,6 @@ def force_refresh_scripts
   $manager.start_scripts
 end
 
-def stop_updating_script(filename)
-  $manager.remove_script(format_name(filename))
-end
```
Comment (rcuhljr) [inline dependency.lic]: Not an option, just use developer mode.

File: dependency.lic [hunk 3]
```diff
@@ -897,15 +879,8 @@ def full_install
   respond 'Finished install!'
 end
 
-def migrate_scripts
-  remove_from_autostart(managed_scripts)
-  add_self_to_autostart
-  managed_scripts.each(&method(:get_script))
-end
```
Comment (rcuhljr) [inline dependency.lic]: Not used anymore.

File: dependency.lic [hunk 4]
```diff
@@ -7,8 +7,9 @@ require 'net/http'
 require 'base64'
 require 'yaml'
 require 'ostruct'
+require 'digest/sha1'
 
-$DEPENDENCY_VERSION = '1.0'
+$DEPENDENCY_VERSION = '1.1'
```
Comment (rpherbig) [inline dependency.lic]: I have a dependency change incoming for bankbot to write to disk and was planning on using 1.1 for that. Just keep an eye out for whichever of these changes lands on master first.

File: dependency.lic [hunk 5]
```diff
@@ -202,6 +203,8 @@ class ScriptManager
     @shop_update_queue = []
 
     @enable_fork = LICH_VERSION =~ /f/
+
+    @versions = nil
```
Comment (rpherbig) [inline dependency.lic]: Why not initialize `@versions = build_version_hash` here?

File: dependency.lic [hunk 6]
```diff
@@ -225,23 +228,21 @@ class ScriptManager
...
-    File.open('script_versions.dat', 'wb') { |file| Marshal.dump(data, file) }
-  end
-
-  def remove_script(filename)
-    data = get_versions
-    data.delete(filename)
-    File.open('script_versions.dat', 'wb') { |file| Marshal.dump(data, file) }
+  def build_version_hash
+    versions = {}
+    Dir['./scripts/*.lic'].map { |item| File.basename(item) }.each do |file|
```
Comment (rpherbig) [inline dependency.lic]: You can use `.each_with_object({})` to simplify this.

**Comment by rpherbig** [comment]
I haven't tested the change itself. Do you want me to update to the branch and... run it for a while?

**Comment by rcuhljr** [comment]
If you don't mind. I haven't tested it either, I wrote it blind. I wont have time to test it until late tonight at the earliest.

File: dependency.lic (hunk 6 above)
Comment (rcuhljr) [inline dependency.lic]: Yeah I was just being lazy and didn't want to look up the syntax.

File: dependency.lic (hunk 5 above)
Comment (rcuhljr) [inline dependency.lic]: No reason, didn't see an advantage one way or another.

Comment (rcuhljr) [inline dependency.lic]: Now that I look at it, doing it this way prevents it from computing hashes for developer mode, but that's a debatable win.

**Pull request merged**